test/Microsoft.ML.Sweeper.Tests/TestSweeper.cs

```diff
@@ -0,0 +1,756 @@
+        }
+
+        [Fact]
+        public void TestLongValueGeneratorRoundTrip()
+        {
+            var paramSweep = new LongValueGenerator(new LongParamArguments() { Name = "bla", Min = 0, Max = 17 });
+            var value = new LongParameterValue("bla", 5);
+            float normalizedValue = paramSweep.NormalizeValue(value);
+            IParameterValue unNormalizedValue = paramSweep.CreateFromNormalized(normalizedValue);
+            Assert.Equal("5", unNormalizedValue.ValueText);
+
+            IParameterValue param = paramSweep.CreateFromNormalized(0.345);
+            float unNormalizedParam = paramSweep.NormalizeValue(param);
+            Assert.Equal("5", param.ValueText);
+            Assert.Equal((Float)5 / 17, unNormalizedParam);
+        }
+
+        [Theory]
+        [InlineData("bla", 10, 20, false, 10, 1000, "15")]
+        [InlineData("bla", 10, 1000, true, 10, 1000, "100")]
+        [InlineData("bla", 10, 1000, true, 10, 3, "100")]
+        public void TestFloatValueSweep(string name, int min, int max, bool logBase, int stepSize, int numSteps, string valueText)
+        {
+            var paramSweep = new FloatValueGenerator(new FloatParamArguments() { Name = name, Min = min, Max = max, LogBase = logBase, StepSize = stepSize, NumSteps = numSteps });
+            IParameterValue value = paramSweep.CreateFromNormalized(0.5);
+            Assert.Equal(name, value.Name);
+            Assert.Equal(valueText, value.ValueText);
+        }
+
+        [Fact]
+        public void TestFloatValueGeneratorRoundTrip()
+        {
+            var paramSweep = new FloatValueGenerator(new FloatParamArguments() { Name = "bla", Min = 1, Max = 5 });
+            var random = new Random(123);
+            var normalizedValue = (Float)random.NextDouble();
+            var value = (FloatParameterValue)paramSweep.CreateFromNormalized(normalizedValue);
+            var originalNormalizedValue = paramSweep.NormalizeValue(value);
+            Assert.Equal(normalizedValue, originalNormalizedValue);
+
+            var originalValue = (FloatParameterValue)paramSweep.CreateFromNormalized(normalizedValue);
+            Assert.Equal(originalValue.Value, value.Value);
+        }
+
+        [Theory]
+        [InlineData(0.5, "bar")]
+        [InlineData(0.75, "baz")]
+        public void TestDiscreteValueSweep(double normalizedValue, string expected)
+        {
+            var paramSweep = new DiscreteValueGenerator(new DiscreteParamArguments() { Name = "bla", Values = new[] { "foo", "bar", "baz" } });
+            var value = paramSweep.CreateFromNormalized(normalizedValue);
+            Assert.Equal("bla", value.Name);
+            Assert.Equal(expected, value.ValueText);
+        }
+
+        [Fact]
+        public void TestRandomSweeper()
+        {
+            var args = new SweeperBase.ArgumentsBase();
+            using (var env = new TlcEnvironment(42))
+            {
+                args.SweptParameters = new[]
```

Comment:
(nit) - can just new up `args` on this line.  No need to put the new `SweeperBase.ArgumentsBase()` before the `new TlcEnvironment`